apps/opik-backend/src/main/java/com/comet/opik/domain/LlmProviderApiKeyService.java

```diff
@@ -0,0 +1,118 @@
+package com.comet.opik.domain;
+
+import com.comet.opik.api.ProviderApiKey;
+import com.comet.opik.api.ProviderApiKeyUpdate;
+import com.comet.opik.api.error.EntityAlreadyExistsException;
+import com.comet.opik.api.error.ErrorMessage;
+import com.google.inject.ImplementedBy;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.core.Response;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
+import ru.vyarus.guicey.jdbi3.tx.TransactionTemplate;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.List;
+import java.util.UUID;
+
+import static com.comet.opik.infrastructure.db.TransactionTemplateAsync.READ_ONLY;
+import static com.comet.opik.infrastructure.db.TransactionTemplateAsync.WRITE;
+
+@ImplementedBy(LlmProviderApiKeyServiceImpl.class)
+public interface LlmProviderApiKeyService {
+
+    ProviderApiKey get(UUID id, String workspaceId);
+    ProviderApiKey saveApiKey(ProviderApiKey providerApiKey, String userName, String workspaceId);
+    void updateApiKey(UUID id, ProviderApiKeyUpdate providerApiKeyUpdate, String userName, String workspaceId);
+}
+
+@Slf4j
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+class LlmProviderApiKeyServiceImpl implements LlmProviderApiKeyService {
+
+    private static final String PROVIDER_API_KEY_ALREADY_EXISTS = "Api key for this provider already exists";
+    private final @NonNull IdGenerator idGenerator;
+    private final @NonNull TransactionTemplate template;
+
+    @Override
+    public ProviderApiKey get(UUID id, String workspaceId) {
+        log.info("Getting provider api key with id '{}', workspaceId '{}'", id, workspaceId);
```

Comment:
Let's be consistent with logs. We can use them either in all public methods or just as controllers.